### PR TITLE
ISPN-11076 ParticipantFailsAfterPrepareTest.testNonOriginatorFailsAft…

### PR DESCRIPTION
**Pull request by danberindei**
…erPrepare random failures

https://issues.redhat.com/browse/ISPN-11076

### CODE REVIEW

File: core/src/test/java/org/infinispan/test/MultipleCacheManagersTest.java [hunk 1]
```diff
@@ -775,6 +775,12 @@ protected Address address(Cache<?, ?> c) {
       return c.getAdvancedCache().getRpcManager().getAddress();
    }
 
+   protected List<Address> addresses() {
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/test/MultipleCacheManagersTest.java]: unused method

File: core/src/test/java/org/infinispan/tx/ParticipantFailsAfterPrepareTest.java [hunk 2]
```diff
@@ -24,81 +27,77 @@
...
 
-   private List<Cache> getAliveParticipants(int indexToKill) {
-      List<Cache> participants = new ArrayList<>();
-      for (int i = 0; i < 4; i++) {
-         if (i == indexToKill) continue;
-         participants.add(cache(i));
+      //now check whether all caches have the same content and no locks acquired
+      for (Cache<?, ?> c : caches()) {
+         // If the new owner becomes the primary, it will release the key asynchronously
+         eventuallyEquals(false, () -> extractLockManager(c).isLocked(key));
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/tx/ParticipantFailsAfterPrepareTest.java]: `assertEventuallyNotLocked(c, key)`?

**Comment by danberindei** [comment]
Updated @pruivo

**Comment by pruivo** [comment]
integrated! thanks @danberindei !

**Pull request merged**